puppet/zulip_ops/templates/iptables/rules.v4.erb

```diff
@@ -40,9 +40,14 @@
 -A INPUT -p tcp --dport https      -j ACCEPT
 -A INPUT -p tcp --dport postgresql -j ACCEPT
 
+<% if @fqdn.include? "smokescreen" -%>
```

Comment:
I'd forgotten we use this approach for templating.  Would probably be better to switch these conditionals to go from the set of enabled puppet rules.

Reply:
Yeah.  The Most Right solution is to use something like https://forge.puppet.com/modules/puppetlabs/firewall so we can declare the iptables rules adjacent to the services that use them.  We currently don't have any story for using or maintaining puppet modules, but AFAICT that's just because we haven't invested in doing so yet.

Reply:
Yeah, we used to have a few third-party modules checked into the codebase under `puppet/`, but ended up removing them because they're kinda a pain -- we'd have problems where one couldn't upgrade a module because it starting requiring a newer version of Ruby, etc.  After a bit of investigation, we determined they hardly did anything for us, and just got rid of them all.  The situation may have changed since then, but in general I just wasn't super impressed by the quality of the puppet modules ecosystem.

In terms of what's Most Right -- that certainly is the Puppet Labs way of doing things, and probably does make sense if you are managing a large fleet of systems.  But probably the single file with a template will remain a more readable way to do what we're trying to do than using a special Puppet module.  (I do think there's significant value in our configuration being readable for someone who doesn't know puppet, since that's the case for most of our contributors and users).

And in this case, https://forge.puppet.com/modules/puppetlabs/firewall is basically a custom language that one needs to learn on top of the normal `iptables` language; at least for me, I'd rather just have an iptables config file and not need to learn that extra language.